go/enclave/rpc/EstimateGas.go

```diff
@@ -94,7 +93,7 @@ func EstimateGasExecute(builder *CallBuilder[CallParamsWithBlock, hexutil.Uint64
 	// TODO: Change to fixed time period quotes, rather than this.
 	publishingGas = publishingGas.Mul(publishingGas, gethcommon.Big2)
 
-	executionGasEstimate, gasPrice, err := rpc.doEstimateGas(builder.ctx, txArgs, blockNumber, rpc.config.GasLocalExecutionCapFlag)
+	executionGasEstimate, gasPrice, err := rpc.estimateGasSinglePass(builder.ctx, txArgs, blockNumber, rpc.config.GasLocalExecutionCapFlag)
```

Comment:
needs a high-level comment of the approach maybe with a simple example.
I find it quite hard to follow what's going on. This stuff is a bit confusing

Reply:
It needs a bit of refactoring tbh, but I guess we can go with a comment for now.

Reply:
Added some comments in a couple of places.